pkg/query-service/app/opamp/logspipeline.go

```diff
@@ -0,0 +1,223 @@
+		if err != nil {
+			return confHash, err
+		}
+		if p.Pipelines.Logs == nil {
+			return confHash, fmt.Errorf("logs pipeline doesn't exist")
+		}
+
+		// build the new processor list
+		updatedProcessorList, _ := buildLogsProcessors(p.Pipelines.Logs.Processors, parsingProcessorsNames)
+		p.Pipelines.Logs.Processors = updatedProcessorList
+
+		// add the new processor to the data ( no checks required as the keys will exists)
+		c["service"].(map[string]interface{})["pipelines"].(map[string]interface{})["logs"] = p.Pipelines.Logs
+
+		updatedConf, err := yaml.Parser().Marshal(c)
+		if err != nil {
+			return confHash, err
+		}
+
+		// zap.S().Infof("sending new config", string(updatedConf))
+		hash := sha256.New()
+		_, err = hash.Write(updatedConf)
+		if err != nil {
+			return confHash, err
+		}
+		agent.EffectiveConfig = string(updatedConf)
+		err = agent.Upsert()
+		if err != nil {
+			return confHash, err
+		}
+
+		agent.SendToAgent(&protobufs.ServerToAgent{
+			RemoteConfig: &protobufs.AgentRemoteConfig{
+				Config: &protobufs.AgentConfigMap{
+					ConfigMap: map[string]*protobufs.AgentConfigFile{
+						"collector.yaml": {
+							Body:        updatedConf,
+							ContentType: "application/x-yaml",
+						},
+					},
+				},
+				ConfigHash: hash.Sum(nil),
+			},
+		})
+
+		if confHash == "" {
+			confHash = string(hash.Sum(nil))
+			model.ListenToConfigUpdate(agent.ID, confHash, callback)
+		}
+	}
+
+	return confHash, nil
+}
+
+// check if the processors already exist
+// if yes then update the processor.
+// if something doesn't exists then remove it.
+func buildLogParsingProcessors(agentConf, parsingProcessors map[string]interface{}) error {
+	agentProcessors := agentConf["processors"].(map[string]interface{})
+	exists := map[string]struct{}{}
+	for key, params := range parsingProcessors {
+		agentProcessors[key] = params
+		exists[key] = struct{}{}
+	}
+	// remove the old unwanted processors
+	for k := range agentProcessors {
+		if _, ok := exists[k]; !ok && strings.HasPrefix(k, constants.LogsPPLPfx) {
+			delete(agentProcessors, k)
+		}
+	}
+	agentConf["processors"] = agentProcessors
+	return nil
+}
+
+type otelPipeline struct {
+	Pipelines struct {
+		Logs *struct {
+			Exporters  []string `json:"exporters" yaml:"exporters"`
+			Processors []string `json:"processors" yaml:"processors"`
+			Receivers  []string `json:"receivers" yaml:"receivers"`
+		} `json:"logs" yaml:"logs"`
+	} `json:"pipelines" yaml:"pipelines"`
+}
+
+func getOtelPipelinFromConfig(config map[string]interface{}) (*otelPipeline, error) {
+	if _, ok := config["service"]; !ok {
+		return nil, fmt.Errorf("service not found in OTEL config")
+	}
+	b, err := json.Marshal(config["service"])
+	if err != nil {
+		return nil, err
+	}
+	p := otelPipeline{}
+	if err := json.Unmarshal(b, &p); err != nil {
+		return nil, err
+	}
+	return &p, nil
+}
+
+func buildLogsProcessors(current []string, logsParserPipeline []string) ([]string, error) {
```

Comment:
I know you originally used the same version of sampling rules logic, but does it have to be complicated as this for your case since we are going with the fact all SigNoz added processors would be a single group, and in the order, they are defined in the UI.

Reply:
So even if we consider that everything will be together and processors can be added at the beginning or towards the end of the group, this logic handles them. 

I agree that if we spend some more time we might be able to reduce the complexity where we don't need to consider anything in between but overall the logic will be something similar.